src/queue.ts

```diff
@@ -613,11 +616,13 @@ abstract class BaseQueue<Message>
                   },
                   ResultPath: heapVar,
                   Next: "send message batch",
-                  Iterator: {
-                    StartAt: "serialize Message",
-                    States: {
+                  Iterator: context.aslGraphToStates({
```

Comment:
Why is this necessary? The inside of an iterator is an isolated state machine, no?

Reply:
I think state names need to be globally unique, can add a test, but it also worked better with the new intrinsic API.

Reply:
I would be surprised if that were true.

Reply:
You can't transition to states inside the iterator from outside

Reply:
> States are elements in your state machine. A state is referred to by its name, which can be any string, but which must be unique within the scope of the entire state machine.

Reply:
Did you test it?

Reply:
Looks like sendmessagebatch and receive were broken after one of my last PRs. (there were no tests)

But after fixing that...

```ts
      return new StepFunction(scope, "q", async () => {
        await queue.sendMessage({ MessageBody: { id: "hello" } });
        await queue.sendMessageBatch({
          Entries: [{ MessageBody: { id: "hello" }, Id: "1" }],
        });
        await queue.sendMessageBatch({
          Entries: [{ MessageBody: { id: "hello" }, Id: "2" }],
        });
      });
```

```
InvalidDefinition: Invalid State Machine Definition: 'DUPLICATE_STATE_NAME: Duplicate State name: serialize Message at /States/1__await queue.sendMessageBatch({Entries: [{MessageBody: {id: "hello"}, Id: 1/Iterator/States/serialize Message, DUPLICATE_STATE_NAME: Duplicate State name: unwrap Message at /States/1__await queue.sendMessageBatch({Entries: [{MessageBody: {id: "hello"}, Id: 1/Iterator/States/unwrap Message'
```

Reply:
Cool, good to know. Seems silly to me as you would think scoping rules would apply. Thanks for checking and fixing.

Reply:
np, will add a new test with this PR.